gax/src/main/java/com/google/api/gax/batching/BatcherImpl.java

```diff
@@ -176,6 +177,7 @@ public void onSuccess(ResponseT response) {
           @Override
           public void onFailure(Throwable throwable) {
             try {
+              batcherStats.recordBatchFailure(throwable);
```

Comment:
can we push this down into Batch to make it symmetrical with onBatchSuccess?

Reply:
I actually tried with the same approach, but If some exception occurs while post-processing in `splitResponse()` then we would end up with 1 batch and n partial failures.

https://github.com/googleapis/gax-java/blob/39fb47171bd786133ccdc140bcaf946b6375a322/gax/src/main/java/com/google/api/gax/batching/BatcherImpl.java#L251-L267

I thought that would be a misleading figure, that why I kept the `BatcherStats#recordBatchFailure` here.

Reply:
I think it would be better to keep the code symmetrical and then have a follow up PR that handles failures from splitResponse

Reply:
Thanks for the suggestion, Updated this change with symmetrical code and added a `TODO` as well for `splitResponse()`.